tests/test_text_extraction.py

```diff
@@ -219,3 +219,13 @@ def test_text_leading_height_unit():
     page = reader.pages[0]
     extracted = page.extract_text()
     assert "Something[cited]\n" in extracted
+
+
+def test_layout_mode_vertical_space():
+    """Tests for #2915"""
+    reader = PdfReader(RESOURCE_ROOT / "attachment.pdf")
```

Comment:
I would recommend using `crazyones.pdf` instead which should have the same content.

Apart from this: Could you please extend the test to actually show the difference between different weights? By just looking at the existing test, it is not really clear to me where the actual difference is (unless looking at others tests or `crazyones.txt`).